open-hackathon-client/src/client/static/js/views/oh.manage.hostserver.js

```diff
@@ -0,0 +1,252 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+
+(function($, oh) {
+
+    var currentHackathon = oh.comm.getCurrentHackathon();
+    var isUpdateOperation = false;
+    var updateHostserverId = 0;
+
+    function toggleTable() {
+        if ($('#hostserverlisttable').css('display') == 'none') {
+            $('#hostserverlisttable').show();
+            $('#hostserverform').hide()
+        } else {
+            $('#hostserverlisttable').hide();
+            $('#hostserverform').show()
+        }
+    }
+
+    function validateIp(ip) {
+        var reg = ip.match(/([0-9]{1,3}\.{1}){3}[0-9]{1,3}/);
+        if(reg == null)
+            return false;
+        else
+            return true;
+    }
+
+    function setFormData(data) {
+        $('#vm_name').val(data.vm_name);
+        $('#public_dns').val(data.public_dns);
+        $('#public_ip').val(data.public_ip);
+        $('#public_docker_api_port').val(data.public_docker_api_port);
+        $('#private_ip').val(data.private_ip);
+        $('#private_docker_api_port').val(data.private_docker_api_port);
+        $('#container_max_count').val(data.container_max_count);
+        $('#disabled').val(data.disabled == 0 ? "启用":"禁用");
+    }
+
+    function clearFormData() {
+        $('#vm_name').val("");
+        $('#public_dns').val("");
+        $('#public_ip').val("");
+        $('#public_docker_api_port').val("4243");
+        $('#private_ip').val("");
+        $('#private_docker_api_port').val("4243");
+        $('#container_max_count').val("1");
+    }
+
+    function getFormData() {
+        data = {};
+        data.vm_name = $('#vm_name').val();
+        data.public_dns = $('#public_dns').val();
+        data.public_ip = $('#public_ip').val();
+        data.public_docker_api_port = $('#public_docker_api_port').val();
+        data.private_ip = $('#private_ip').val();
+        data.private_docker_api_port = $('#private_docker_api_port').val();
+        data.container_max_count = parseInt($('#container_max_count').val(), 10);
+        data.disabled = $('#disabled').val() == '启用'? 0:1; //0:abled 1:disabled
+        data.is_auto = 2 //auto=1, manual=2;
+        return data;
+    }
+
+    function createHostServer(data) {
+        alert("该请求需要测试Host Server的Docker服务是否可用,点击确认/ok开始，请等待最多5秒");
```

Comment:
Please don't use `alert` of `confirm` seriously other than debugging.  It's usually blocked by browsers and users. I don't think it's required in `create` or `insert` behaviors. If have to ,please find existing examples in our code or talk with Boli Guan.